src/query/v2/plan/operator.cpp

```diff
@@ -2450,6 +2466,64 @@ class DistributedCreateExpandCursor : public Cursor {
     return edge_requests;
   }
 
+  std::vector<msgs::NewExpand> ExpandCreationInfoToRequests(MultiFrame &multi_frame, ExecutionContext &context) const {
+    std::vector<msgs::NewExpand> edge_requests;
+    auto reader = multi_frame.GetValidFramesModifier();
+
+    for (auto &frame : reader) {
+      const auto &edge_info = self_.edge_info_;
+      msgs::NewExpand request{.id = {context.edge_ids_alloc->AllocateId()}};
+      ExpressionEvaluator evaluator(&frame, context.symbol_table, context.evaluation_context, nullptr,
+                                    storage::v3::View::NEW);
+      request.type = {edge_info.edge_type};
+      if (const auto *edge_info_properties = std::get_if<PropertiesMapList>(&edge_info.properties)) {
+        for (const auto &[property, value_expression] : *edge_info_properties) {
+          TypedValue val = value_expression->Accept(evaluator);
+          request.properties.emplace_back(property, storage::v3::TypedValueToValue(val));
+        }
+      } else {
+        // handle parameter
+        auto property_map = evaluator.Visit(*std::get<ParameterLookup *>(edge_info.properties)).ValueMap();
+        for (const auto &[property, value] : property_map) {
+          const auto property_id = context.request_router->NameToProperty(std::string(property));
+          request.properties.emplace_back(property_id, storage::v3::TypedValueToValue(value));
+        }
+      }
+      // src, dest
+      TypedValue &v1_value = frame[self_.input_symbol_];
+      const auto &v1 = v1_value.ValueVertex();
+      const auto &v2 = OtherVertex(frame);
+
+      // Set src and dest vertices
+      // TODO(jbajic) Currently we are only handling scenario where vertices
+      // are matched
+      const auto set_vertex = [](const auto &vertex, auto &vertex_id) {
+        vertex_id.first = vertex.PrimaryLabel();
+        vertex_id.second = vertex.GetVertex().id.second;
```

Comment:
`GetVertex()` creates a copy, that I don't like. As I haven't noticed that before (if I was a reviewer, sorry for it), but it might cause issues, please update that function to return a `const &`.

But even then, I think we should use `vertex_id = vertex.Id()` which totally eliminates the need for this lambda.